salt/states/file.py

```diff
@@ -4668,23 +4668,23 @@ def manage_directory(path):
         name, source, keep_symlinks, include_pat, exclude_pat, maxdepth, include_empty
     )
 
+    for dirname in mng_dirs:
+        manage_directory(dirname)
+    for dest, src in mng_files:
+        manage_file(dest, src, replace)
     for srelpath, ltarget in mng_symlinks:
         _ret = symlink(
             os.path.join(name, srelpath),
             ltarget,
             makedirs=True,
-            force=force_symlinks,
+            force=True,
```

Comment:
I think this inadvertently got committed. I think `force_symlinks` is still supposed to be used.

Reply:
No, I meant to do this. Since this is happening after dirs and files, it fails because the symlink exists. This updates the symlink.

Reply:
Perhaps this is better:
`force=force_symlinks or keep_symlinks`

Reply:
Though I'm not sure what effect `force_symlinks` has, or what the intended functionality is.

Reply:
I am not sure that keep_symlinks is supposed to force the symlink creation:

```
keep_symlinks
        Keep symlinks when copying from the source. This option will cause
        the copy operation to terminate at the symlink. If desire behavior
        similar to rsync, then set this to True. This option is not taken
        in account if ``fileserver_followsymlinks`` is set to False.

    force_symlinks
        Force symlink creation. This option will force the symlink creation.
        If a file or directory is obstructing symlink creation it will be
        recursively removed so that symlink creation can proceed. This
        option is usually not needed except in special circumstances. This
        option is not taken in account if ``fileserver_followsymlinks`` is
        set to False.
```

Reply:
Well, they aren't symlinks without it.

Reply:
I can't think of a scenario where you would pass `keep_symlinks` and not expect them to be symlinks.